### DISCUSSION TITLE
Allow further filtering of custom props

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#846] [Anything Else]
"Custom Goals.. clicking on a “Breakdown by" prop value -> it activates the filter for this value.

Now I would have expected to still see the other “Breakdown by” categories and be able to filter down further and get insights on how one “Breakdown by” category affects my other “Breakdown by” categories for this goal.

I would really love to see this!"

**Update:** You can now segment your audience by multiple custom properties. Just keep selecting additional custom properties in your dashboard as you’re filtering. 

You can use the filter button on the top right of the dashboard to do the same too. There we’ve also added the “contains” option for even more powerful segmenting.

**Comment by ukutaht** [comment]
Good point. Seems like an arbitrary limitation in the UI currently.

Good use-case example is [elementary.io's dashboard](https://plausible.io/elementary.io). It would make sense to want to see conversions of the `Payment` event filtered by `Action=Complete` and to then see a breakdown of the `Amount`.

**Comment by neuromusic** [comment]
+1 to this request. (see https://github.com/chanzuckerberg/napari-hub/pull/76#issuecomment-858023090 for our use case)

**Reply by ukutaht** [reply]
Cheers. Clickhouse is working hard to add better support for semi-structured data: [Their roadmap for 2021](https://github.com/ClickHouse/ClickHouse/issues/17623).

The `Map` datatype which we need is currently in experimental mode but I hope to see it stable soon. Once it is, I'm planning to migrate our custom prop data to the new datatype which offers more flexibility.

**Reply by UnamedRus** [reply]
Experimental status of map datatype was removed in that pr  https://github.com/ClickHouse/ClickHouse/pull/25773 .
So starting from 21.8, it's not experimental.

But internally it's still 2 arrays, so performance different from two key value arrays shoudn't be big.

**Reply by ukutaht** [reply]
Sweet. I'll probably get around to upgrading in the coming months after it reaches the Altinity stable channel: https://docs.altinity.com/altinitystablerelease/releasenotes/. Then I can start playing around with Maps

**Comment by Wolfkhan66** [comment]
Any update on this? More and more this is becoming an urgent requirement and we now starting to look at alternatives to plausible to fulfill this need

**Reply by ukutaht** [reply]
Current status is that we are waiting for JSON datatype support for Clickhouse which is supposed to be coming [this year](https://github.com/ClickHouse/ClickHouse/issues/32513). The problem with `Map` is that it is must be supplied with a static type pair (e.g. `Map(String, UInt64)`) so it cannot handle mixed types which is an important aspect of custom props.

**Comment by dbstraight** [comment]
Is this potentially coming in 2023?

**Reply by earthboundkid** [reply]
It looks like the blocker from Clickhouse got resolved https://docs.altinity.com/releasenotes/altinity-stable-release-notes/22.8/altinity-stable-22.8.13/. How is this going on the Plausible side @ukutaht? I would like to use Plausible instead of going to GA4 (yuck), but the ability to drill down into custom props is a requirement for us, so we can't switch over without it.

**Reply by ukutaht** [reply]
> How is this going on the Plausible side @ukutaht?

I don't want to make any promises. We are planning to work on the Clickhouse schema later this year so there's a good chance we will migrate the custom props over to the `Map` datatype as well.

**Comment by metmarkosaric** [comment]
**Update:** You can now segment your audience by multiple custom properties. Just keep selecting additional custom properties in your dashboard as you’re filtering. 

You can use the filter button on the top right of the dashboard to do the same too. There we’ve also added the “contains” option for even more powerful segmenting.

**Reply by mir1198yusuf** [reply]
@metmarkosaric I am not able to select 2nd custom property filter if one exists. when i do, it replaces first custom property filter with new one .
<img width="477" alt="Screenshot 2024-05-03 at 12 42 48 PM" src="https://github.com/plausible/analytics/assets/34382235/8518ab1e-e891-4ac2-92d0-ab6c656cc944">

In above image, I want to add a filter for new property `is_admin`, how can i do that ?
